Mathlib/NumberTheory/NumberField/Discriminant.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.NumberTheory.NumberField.Basic
+import Mathlib.RingTheory.Localization.NormTrace
+
+/-!
+# Number field discriminant
+This file defines the discriminant of a number field.
+
+## Main definitions
+ - `discr` the discriminant of a number field.
+
+## Tags
+number field, discriminant
+-/
+
+namespace NumberField
+
+open NumberField Matrix
+
+variable (K : Type*) [Field K] [NumberField K]
+
+/-- The discriminant of a number field. -/
+noncomputable def discr : ℤ := Algebra.discr ℤ (RingOfIntegers.basis K)
+
+theorem coe_discr : (discr K : ℚ) = Algebra.discr ℚ (integralBasis K) := by
+  rw [discr]
+  exact (Algebra.discr_localizationLocalization ℤ _ K (RingOfIntegers.basis K)).symm
+
+theorem discr_ne_zero : discr K ≠ 0 := by
+  rw [← (Int.cast_injective (α := ℚ)).ne_iff, coe_discr]
+  exact Algebra.discr_not_zero_of_basis ℚ (integralBasis K)
+
+theorem discr_eq_discr {ι : Type*} [Fintype ι] [DecidableEq ι] (b : Basis ι ℤ (𝓞 K)) :
+    Algebra.discr ℤ b = discr K := by
+  let b₀ := Basis.reindex (RingOfIntegers.basis K) (Basis.indexEquiv (RingOfIntegers.basis K) b)
+  rw [Algebra.discr_eq_discr (𝓞 K) b b₀, discr, Basis.coe_reindex, Algebra.discr_reindex]
+
+end NumberField
+
+namespace Rat
+
+open NumberField
+
+theorem discr : discr ℚ = 1 := by
```

Comment:
```suggestion
theorem discr_eq_one : discr ℚ = 1 := by
```

Reply:
Since there are already [Rat.numberField](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/Basic.html#Rat.numberField) and [Rat.ringOfIntegersEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/Basic.html#Rat.ringOfIntegersEquiv), maybe the best compromise is `Rat.numberField_discr`?